File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/exercicios/Contador.java
import java.util.*;

public class Contador{
    public static int IsFim(String line){
        int result = 1;

        if(line.charAt(0) == 'F' && line.charAt(1) == 'I' && line.charAt(2) == 'M'){
            result = 0;
        }

        return result;
    }

    public static int Verificar(char c){
        int result = 0;

        if(c >= 'A' && c <= 'Z'){
            result = 1;
        }

        return result;
    }

    public static int Contar(String line){
        int count = 0;

        for(int i = 0; i < line.length(); i++){
            if(Verificar(line.charAt(i)) == 1){
                count++;
            } 
        }

        return count;
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String line;

        line = sc.nextLine();
        while(IsFim(line) != 0){
            System.out.println(Contar(line));
            line = sc.nextLine();
        }

        sc.close();
    }
}

File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/exercicios/Contador.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int IsFim(char line[]){
    int result = 1;

    if(line[0] == 'F' && line[1] == 'I' && line[2] == 'M'){
        result = 0;
    }

    return result;
}

int Contar(char line[]){
    int count = 0;

    for(int i = 0; i < strlen(line); i++){
        char c = line[i];

        if(line[i] == '\0'){ 
            count = 0;
        } else {
            if(c >= 'A' && c <= 'Z'){  
                count++;
            }
        }
    }

    return count; 
}

int main(){
    char line[100];

    scanf("%[^\n]", line);
    while(IsFim(line) != 0){
        printf("%d\n", Contar(line));
        scanf(" %[^\n]", line);
    }
}
